### PR TITLE
Simplifying ExpectationsBuilder.go

### PR DESCRIPTION
**Pull request by jglick**
As discussed in https://github.com/jenkinsci/workflow-job-plugin/pull/131#discussion_r292131627, of which this is downstream.

### CODE REVIEW

**Comment by jglick** [comment]
Note that while this PR includes incrementalification of the plugin, actual deployment would work only if #335 were also merged.

**Comment by dwnusbaum** [comment]
[`ToolsTest.buildPluginParentPOM`](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/1fa15e95b45f064c5639da86ef23d5cbc4ce31be/pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/ToolsTest.java#L102) failed in the CI build on the Java 11 Linux branch with this message:

> Expected: a string containing "Finished: SUCCESS, but was ... (build log ends with Finished: FAILURE)"

It looks like some kind of Maven failure:

```
[ERROR] Could not create local repository at /home/jenkins/.m2/repository -> [Help 1]
org.apache.maven.repository.LocalRepositoryNotAccessibleException: Could not create local repository at /home/jenkins/.m2/repository
at org.apache.maven.DefaultMaven.validateLocalRepository(DefaultMaven.java:368)
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:171)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/LocalRepositoryNotAccessibleException
```

**Comment by jglick** [comment]
I swear I just saw this same error somewhere else…

File: pom.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
       <dependency>
         <groupId>org.jenkins-ci.plugins.workflow</groupId>
         <artifactId>workflow-job</artifactId>
-        <version>2.26</version>
+        <version>2.33-rc970.e82d4fa96516</version> <!-- TODO https://github.com/jenkinsci/workflow-job-plugin/pull/131 -->
```
Comment (dwnusbaum) [inline pom.xml]:
```suggestion
        <version>2.33</version>
```

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/AbstractModelDefTest.java [hunk 2]
```diff
@@ -500,37 +500,9 @@ public WorkflowRun go() throws Exception {
...
-                */
-                System.out.println("Build result did not initially match " + result);
-                Thread.sleep(1000);
-                j.assertBuildStatus(result, run);
-                System.out.println("But after 1s build result _did_ match " + result);
-            }
-            // To deal with some erratic failures due to error logs not showing up until after "completion"
-            Thread.sleep(100);
+            j.waitForCompletion(run);
+            j.waitForMessage("Finished: " + result, run); // like BuildListener.finished
```
Comment (dwnusbaum) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/AbstractModelDefTest.java]: Maybe worth noting why this line exists with something like this so no one wastes time wondering what the redundancy is for:
```suggestion
            // Calling `j.assertBuildStatus` directly after `j.waitForCompletion` is subject to race conditions in Pipeline jobs, so the call to `j.waitForMessage` is just here to ensure that the build is totally complete before checking the status.
            // If that race condition is fixed, the call to `j.waitForMessage` can be removed.
            j.waitForMessage("Finished: " + result, run); // like BuildListener.finished. 
```

**Comment by dwnusbaum** [comment]
Another `ToolsTest.buildPluginParentPOM` failure, rebuilding.

**Comment by jglick** [comment]
No idea what [this failure](https://ci.jenkins.io/job/Plugins/job/pipeline-model-definition-plugin/job/PR-336/8/testReport/junit/org.jenkinsci.plugins.pipeline.modeldefinition.steps/CredentialWrapperStepTest/windows_11_2_164_1___Archive__windows_11_2_164_1____org_jenkinsci_plugins_pipeline_modeldefinition_steps_CredentialWrapperStepTest/) was about.

**Pull request merged**